project/ConsoleTracker/ConsoleTrackerApp/InternalSocketListener.py

```diff
@@ -19,16 +19,31 @@ def listen():
         while True:
             data = s.recv(1024)
             if data:
+                print(data)
                 data = json.loads(data.decode())
-                messageType = int(data["messageType"])
+                msg = data["msg"] if "msg" in data else ""
                 # Check for message type
-                if messageType == 1:  # Update time request
+                if msg == "timer_update":  # Update time request
                     timeBalance = int(data["timeBalance"])
                     username = data["username"]
-                    print("Update time request")
-                    update_user_time(username, timeBalance)
-                elif messageType == 2:  # Ping request
+
+                    if update_user_time(username, timeBalance):
+                        print("Update time request SUCCESS")
+                    else:
+                        print("Update time request FAILED")
+                elif msg == "ping":  # Ping request
                     print("Ping request")
+                    machine_list = json.dumps({
+                        "machines": [dump(machine) for machine in ping_all_machines()],
+                        "dest": "external"
+                    }).encode()
+                    s.sendall(machine_list)
```

Comment:
@digengill this line sends the machine list back to the same connection and restarts this loop because of the line you define **initMsg = {"conn_type": 1}** making only the django side on the receiving end

Reply:
The ServerSocket script has logic to figure out how to send the response to the connection where the request originally came from. In this case, dummyPingRequest needs to send the request, and wait for a response right after that. So you shouldn't have to do anything other than making sure that the django listener does respond.

Reply:
Then in this case is that logic doing something unintended then? Because the response isn't being sent to the ping request but instead it's being sent to the listener which I can't figure out why

Reply:
Yea If thats the case, can you provide the steps I need to do and I can fix what is broken.

Reply:
Yeah just follow the steps I listed above. Run server script then run python runserver then try running the dummy ping request script. After you do that you'll see that the message ends up being received by the listener.